tests/test_qutrit_device.py

```diff
@@ -0,0 +1,670 @@
+        m.setattr(QutritDevice, "density_matrix", lambda self, wires=None: 0)
+        m.setattr(QutritDevice, "probability", lambda self, wires=None, *args, **kwargs: 0)
+        m.setattr(QutritDevice, "apply", lambda self, x, **kwargs: x)
+
+        def get_qutrit_device(wires=1):
+            return QutritDevice(wires=wires)
+
+        yield get_qutrit_device
+
+
+@pytest.fixture(scope="function")
+def mock_qutrit_device_shots(monkeypatch):
+    """A function to create a mock device that mocks the methods related to
+    statistics (expval, var, sample, probability)"""
+    with monkeypatch.context() as m:
+        m.setattr(QutritDevice, "__abstractmethods__", frozenset())
+        m.setattr(QutritDevice, "_capabilities", mock_qutrit_device_capabilities)
+        m.setattr(QutritDevice, "short_name", "MockQutritDevice")
+        m.setattr(QutritDevice, "operations", ["QutritUnitary", "Identity"])
+        m.setattr(QutritDevice, "observables", ["Identity"])
+        m.setattr(QutritDevice, "expval", lambda self, *args, **kwargs: 0)
+        m.setattr(QutritDevice, "var", lambda self, *args, **kwargs: 0)
+        m.setattr(QutritDevice, "sample", lambda self, *args, **kwargs: 0)
+        m.setattr(QutritDevice, "state", 0)
+        m.setattr(QutritDevice, "density_matrix", lambda self, wires=None: 0)
+        m.setattr(QutritDevice, "apply", lambda self, x, **kwargs: x)
+        m.setattr(
+            QutritDevice,
+            "analytic_probability",
+            lambda self, wires=None: QutritDevice.marginal_prob(self, np.ones(9) / 9.0, wires),
+        )
+
+        def get_qutrit_device(wires=1, shots=None):
+            return QutritDevice(wires=wires, shots=shots)
+
+        yield get_qutrit_device
+
+
+@pytest.fixture(scope="function")
+def mock_qutrit_device_with_original_statistics(monkeypatch):
+    """A function to create a mock qutrit device that uses the original statistics related
+    methods"""
+    with monkeypatch.context() as m:
+        m.setattr(QutritDevice, "__abstractmethods__", frozenset())
+        m.setattr(QutritDevice, "_capabilities", mock_qutrit_device_capabilities)
+        m.setattr(QutritDevice, "short_name", "MockQutritDevice")
+        m.setattr(QutritDevice, "operations", ["QutritUnitary", "Identity"])
+        m.setattr(QutritDevice, "observables", ["Identity"])
+
+        def get_qutrit_device(wires=1):
+            return QutritDevice(wires=wires)
+
+        yield get_qutrit_device
+
+
+# TODO: Add tests for expval, var after observables are added
+
+
+class TestOperations:
+    """Tests the logic related to operations"""
+
+    def test_op_queue_accessed_outside_execution_context(self, mock_qutrit_device):
+        """Tests that a call to op_queue outside the execution context raises the correct error"""
+
+        with pytest.raises(
+            ValueError, match="Cannot access the operation queue outside of the execution context!"
+        ):
+            dev = mock_qutrit_device()
```

Comment:
The device creation should be moved outside of the `pytest.raises` context because right now the test passes both when:
* The device creation or
* The `dev.op_queue` call

raises the error that is being checked for. However, we'd like to test that for sure `dev.op_queue` raises this error.

This type of change applies to two other tests too.